src/error.rs

```diff
@@ -32,4 +45,56 @@ pub enum DatabaseError {
     WriteError(String),
     #[error("Failed to delete {0} from the Database database")]
     DeleteError(String),
-}
\ No newline at end of file
+}
+
+#[derive(Error, Debug)]
+pub enum DataAvailabilityError {
+    #[error("Failed to establish WebSocket connection: {0}")]
+    WebSocketError(String),
+    #[error("Namespace initialization error")]
+    NamespaceInitializationError,
+    #[error("The data channel has been closed")]
+    ChannelClosed,
+    #[error("Network error: {0}")]
+    NetworkError(String),
+    #[error("Data retrieval error at height {0}: {1}")]
+    DataRetrievalError(u64, String),
+    #[error("Error parsing data at height {0}")]
+    ParsingError(u64),
+    #[error("Error submitting data at height {0}: {1}")]
+    WriteError(u64, String),
+    #[error("Error sending message to channel")]
+    ChannelError,
```

Comment:
This makes sense to us but it may be confusing when people are trying to figure out how things work. This channel is actually there for receiving new block headers (to set the sync target to) from the DA layer. 

We don't really neeed to change this now until we find a good name but maybe something like BlockSubscriptionError or SyncTargetError works. And it should probably cover both the sending and receiving case, maybe with a more specific error message with details passed in?

#[error("Error from block header subscription channel: {0}")]
BlockSubscriptionError,

or 

#[error("Error updating sync target from channel: {0}")]
SyncTargetError,

wdyt?

Reply:
SyncTarget would be a little less specific or what do you think? We could do something like: 
`#[error("Sync target error during {action}: {message}")]
SyncTargetError`
where action can be replaced by "sending" or "receiving", and message provides additional details about the error. or something like that.. wdyt?

Reply:
Yeah, that sounds great

The only thing I would change is instead of "Sync target error" -> "Error {action} new sync target: {message}"